Source/tess.c

```diff
@@ -382,6 +381,100 @@ int tessMeshTessellateInterior( TESSmesh *mesh )
 			if ( !tessMeshTessellateMonoRegion( mesh, f ) ) return 0;
 		}
 	}
+	return 1;
+}
+
+
+struct EdgeStackNode {
+	TESShalfEdge *edge;
+	struct EdgeStackNode *next;
+};
+
+struct EdgeStack {
+	struct EdgeStackNode *top;
+};
+
+void stackInit(struct EdgeStack *stack)
+{
+    stack->top = NULL;
+}
+
+int stackEmpty(struct EdgeStack *stack)
+{
+	return stack->top == NULL;
+}
+
+void stackPush(struct EdgeStack *stack, TESShalfEdge *e)
```

Comment:
Should use bucket allocator here. Something like this:
```C
struct EdgeStack {
	struct EdgeStackNode *top;
	struct BucketAlloc* nodeBucket;
};

void stackInit( struct  EdgeStack *stack, TESSalloc* alloc )
{
	stack->top = NULL;
	stack->nodeBucket = createBucketAlloc( alloc, “CDT nodes”, sizeof(struct EdgeStackNode), 512 );
	if ( stack->nodeBucket == NULL ) return 0;
	return 1;
}

void stackDelete( struct  EdgeStack *stack )
{
	deleteBucketAlloc( stack-> nodeBucket );
}

void stackPush( struct EdgeStack *stack, TESShalfEdge *e )
{
	struct EdgeStackNode *node = (struct EdgeStackNode *)bucketAlloc( stack->nodeBucket );
	if ( ! node ) return;
	node->edge = e;
	node->next = stack->top;
	stack->top = node;
}

TESShalfEdge *stackPop( struct EdgeStack *stack )
{
	TESShalfEdge *e = NULL;
	struct EdgeStackNode *node = stack->top;
	if (node) {
		stack->top = node->next;
		e = node->edge;
		bucketFree( stack->nodePool, node );
	}
	return e;
}
```
You could use typedefs for the structs, even most of the internal ones use it. I would prefer that, though I noticed the allocator does not use typedef (it should).

In addition `tessMeshRefineDelaunay()` needs TESSalloc passed to it (member of `TESStesselator`), `stackDelete()`  should be called at the end of the function. Custom allocators are allowed to return NULL, we should cope with that.